src/codegen/compile.ml

```diff
@@ -10249,8 +10660,8 @@ and compile_prim_invocation (env : E.t) ae p es at =
   | GetPastArrayOffset spacing, [e] ->
     let shift =
       match spacing with
-      | ElementSize -> compile_shl_const 2l (* effectively a multiplication by word_size *)
-      | One -> BigNum.from_word30 env in    (* make it a compact bignum *)
+      | ElementSize
+      | One -> BigNum.from_word30 env in    (* make it a compact bignum *) (* TBR: from_word30? *)
```

Comment:
If I am not mistaken, it would be the compact integer value with 29 bit payload, excluding the sign bit and the tag...

Reply:
Fixed in #4375